libsolidity/analysis/TypeChecker.cpp

```diff
@@ -532,9 +551,33 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 		{
 			solAssert(!result.message().empty(), "Expected detailed error message");
 			m_errorReporter.typeError(1534_error, _variable.location(), result.message());
+			return false;
 		}
 	}
 
+	if (_variable.isStateVariable())
```

Comment:
Do we need this check?

Reply:
It could instead just be "if reference type of storage location" - do you mean that?

Reply:
Something like that, but I'm not entirely sure. Should the following contract cause the makes-collisions-likely warning?
```
contract C {
    mapping(uint => uint[2**100]) x;
}
```

Currently the warning is reported, but we do not get into the `if (auto referenceType = dynamic_cast<ReferenceType const*>(varType))` branch.

Reply:
Ok, hadn't thought about mappings not being reference types...

Reply:
``if (varType->dataStoredIn(DataLocation::Storage))`` might do the trick...

Reply:
Yep, seems to work!